src/connection.rs

```diff
@@ -205,11 +205,40 @@ pub struct Msg {
 }
 
 impl ActualConnection {
-    pub fn new(addr: &ConnectionAddr) -> RedisResult<ActualConnection> {
+    pub fn new(addr: &ConnectionAddr, timeout: Option<Duration>) -> RedisResult<ActualConnection> {
         Ok(match *addr {
             ConnectionAddr::Tcp(ref host, ref port) => {
                 let host: &str = &*host;
-                let tcp = TcpStream::connect((host, *port))?;
+                let tcp = match timeout {
+                    None => TcpStream::connect((host, *port))?,
+                    Some(timeout) => {
+                        let mut tcp = None;
+                        let mut last_error = None;
+                        for addr in ((host, *port)).to_socket_addrs()? {
```

Comment:
What's the reason to resolve and iterate over the addresses here?

Reply:
This is because host can resolve to multiple IP addresses.
The same process is implicitly done in `TcpStream::connect` function, but `TcpStream::connect_timeout` takes only a single `SocketAddr` as described in [TcpStream::connect_timeout](https://doc.rust-lang.org/std/net/struct.TcpStream.html#method.connect_timeout).

Reply:
That makes sense, thanks!